src/User/AutoLoginMiddleware.php

```diff
@@ -1,16 +1,91 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Yiisoft\Yii\Web\User;
 
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Log\LoggerInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+
 /**
  * AutoLoginMiddleware automatically logs user in based on "remember me" cookie
  */
-class AutoLoginMiddleware
+final class AutoLoginMiddleware implements MiddlewareInterface
 {
     private User $user;
+    private IdentityRepositoryInterface $identityRepository;
+    private LoggerInterface $logger;
 
-    public function __construct(User $user)
-    {
+    public function __construct(
+        User $user,
+        IdentityRepositoryInterface $identityRepository,
+        LoggerInterface $logger
+    ) {
         $this->user = $user;
+        $this->identityRepository = $identityRepository;
+        $this->logger = $logger;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $response = $handler->handle($request);
+        if (!$this->authenticateUserFromRequest($request, $response)) {
+            $this->logger->warning('Unable to authenticate user by cookie.');
+        }
+
+        return $response;
+    }
+
+    /**
+     * Parse and determines if an identity cookie has a valid format.
+     * @param ServerRequestInterface $request Request to handle
+     * @return array Returns an array of 'identity' and 'duration' if valid, otherwise [].
+     */
+    private function parseCredentials(ServerRequestInterface $request): array
+    {
+        try {
+            $cookies = $request->getCookieParams();
+            $data = json_decode($cookies['remember'], true, 512, JSON_THROW_ON_ERROR);
+        } catch (\Exception $e) {
```

Comment:
JsonException?

Reply:
I use exception to also encompass getCookieParams. It is not necessary?

Reply:
Ok